src/redis-cli.c

```diff
@@ -763,6 +764,22 @@ static void freeHintsCallback(void *ptr) {
  * Networking / parsing
  *--------------------------------------------------------------------------- */
 
+/* Unquote a null-terminated string and return it as a binary-safe sds. */
+static sds unquoteCString(char *str) {
+    int count;
+    sds *unquoted = sdssplitargs(str, &count);
+
+    if (unquoted && count == 1) {
+        sds res = unquoted[0];
+        sds_free(unquoted);
```

Comment:
maybe it would be nicer and less fragile to nullify `unquoted[0]` and call `sdsfreesplitres`